cli_tools/common/utils/param/fixer_test.go

```diff
@@ -0,0 +1,379 @@
+//  Copyright 2020 Google Inc. All Rights Reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package param
+
+import (
+	"fmt"
+	"testing"
+
+	"cloud.google.com/go/storage"
+	"github.com/GoogleCloudPlatform/compute-image-tools/daisy"
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/mocks"
+)
+
+func TestFixer_PopulateMissingParametersReturnsErrorWhenZoneCantBeRetrieved(t *testing.T) {
+	project := "a_project"
+	scratchBucketGcsPath := "gs://scratchbucket/scratchpath"
+	zone := ""
+	region := ""
+	file := "gs://a_bucket/a_file"
+	storageLocation := "US"
+
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	mockMetadataGce := mocks.NewMockMetadataGCEInterface(mockCtrl)
+	mockScratchBucketCreator := mocks.NewMockScratchBucketCreatorInterface(mockCtrl)
+	mockResourceLocationRetriever := mocks.NewMockResourceLocationRetrieverInterface(mockCtrl)
+	mockResourceLocationRetriever.EXPECT().GetZone("us-west2", project).Return("",
+		daisy.Errf("zone not found")).Times(1)
+	mockStorageClient := mocks.NewMockStorageClientInterface(mockCtrl)
+	mockStorageClient.EXPECT().GetBucketAttrs("scratchbucket").Return(&storage.BucketAttrs{Location: "us-west2"}, nil).Times(1)
+
+	err := fixer{
```

Comment:
Call NewFixer instead of fixer{} to test one more layer?

Reply:
Nice idea! Done.